docs/source/app_developers_guide/adding_a_node.rst

```diff
@@ -0,0 +1,373 @@
+   from your host system:
+
+   .. code-block:: console
+
+      user@host$ docker-compose -f sawtooth-default-poet.yaml down
+
+
+Ubuntu: Configure and Start a Multiple-node Sawtooth Network
+=============================================================
+
+
+Prerequisites
+-------------
+
+This procedure assumes that you have a working (runnable) validator node that
+has been configured as described in :doc:`ubuntu`.
+
+Step 1: Configure the Network on the First Node
+-----------------------------------------------
+
+#. If the first validator node is running, stop all Sawtooth components as
+   described in :ref:`stop-sawtooth-ubuntu-label`.
+
+#. Open a new terminal window (the validator-registry window) and start the
+   Validator Registry transaction processor,
+   ``sawtooth-poet-validator-registry-tp``.
+
+   .. code-block:: console
+
+      user@validator-registry$ sudo -u sawtooth sawtooth-poet-validator-registry-tp -v
+      [23:47:06 INFO    core] register attempt: OK
+
+#. Configure the Sawtooth settings to allow transactions from the Settings,
+   IntegerKey, XO, and Validator Registry transaction families. Run this
+   command:
+
+   .. code-block:: console
+
+      user@client$ sawset proposal create \
+      sawtooth.validator.transaction_families='[{"family": "intkey", "version": "1.0"},\
+      {"family":"sawtooth_settings", "version":"1.0"}\
+      {"family":"xo", "version":"1.0"}\
+      {"family":"sawtooth_validator_registry", "version":"1.0"}]'
+
+#. Create the ``validator.toml`` configuration file on the first validator node.
+   Use the following command to copy the example configuration file to the
+   config directory, ``/etc/sawtooth``.
+
+   .. code-block:: console
+
+      user@valnode1$ sudo cp /sawtooth-core/packaging/validator.toml.example /etc/sawtooth/validator.toml
+
+   .. note::
+
+      This environment uses the default location and name for this configuration
+      file. For more information, see
+      :doc:`../sysadmin_guide/configuring_sawtooth/validator_configuration_file`.
+
+#. Edit this file to change the following settings:
+
+   a. Set the ``bind`` directive to the network and component endpoints for this
+      node.
+
+      * ``network`` specifies the validator-to-validator connection (by default,
+         ``127.0.0.1:8800``
+
+      * ``component`` specifies the port on which the validator listens for
+         communication from other Sawtooth components (the REST API and
+         transaction processors)
+
+      For this application development environment, you can use the default
+      settings:
+
+      .. code-block:: none
+
+         bind = [
+           "network:tcp://127.0.0.1:8800",
+           "component:tcp://127.0.0.1:4004"
+         ]
+
+   #. Set ``endpoint`` to the IP address and port of the advertised network
+      endpoint. This is the public address that clients will use to contact this
+      validator node via the REST API.  The default value is the same as the
```

Comment:
`endpoint` is the publicly addressable IP and port of the validator to validator connection. it may differ from the `network` setting in `bind` when the validator is hosted behind a firewall. In that case one would be relying on the firewall to map the `endpoint` IP:Port to the `bind` `network` IP:Port.  `endpoint` is the value this validator will share with other validators when it is peering with them.

Reply:
Thanks, Dan.  I'll add this info and fix the port typos that you noted below (plus other egregious typos asap).